tests/unit/commands/local/lib/test_cfn_base_provider.py

```diff
@@ -0,0 +1,28 @@
+from unittest import TestCase
+from unittest.mock import Mock, patch
+
+from samcli.commands.local.lib.swagger.parser import SwaggerParser
+from samcli.lib.providers.cfn_base_api_provider import CfnBaseApiProvider
+
+
+class TestDisableAuthorizerFlagInExtractSwaggerRoute(TestCase):
+
+    @patch("samcli.lib.providers.cfn_base_api_provider.LOG.debug")
+    @patch("samcli.lib.providers.cfn_base_api_provider.SwaggerParser")
+    @patch("samcli.lib.providers.cfn_base_api_provider.SwaggerReader")
+    def test_disable_authorizer_flag_when_enabled_skips_authorizer_extraction(self, mock_reader: Mock, mock_parser: Mock, mock_debug_log: Mock):
+        mock_collector = Mock()
+
+        CfnBaseApiProvider.extract_swagger_route(stack_path="", logical_id="", body={}, uri="", binary_media=None, collector=mock_collector, cwd="", event_type="HttpApi", disable_authorizer=True)
+        assert "Found '%s' authorizers in resource" not in mock_debug_log.call_args[0]
```

Comment:
Could we update this test to instead check that the `mock_collector.add_authorizers` was not called? This might be a more concrete way of testing if the collection logic was not called instead of checking for log calls.